kotlin-mbedtls-netty/src/main/kotlin/org/opencoap/ssl/netty/DtlsChannelHandler.kt

```diff
@@ -90,6 +90,16 @@ class DtlsChannelHandler @JvmOverloads constructor(
 
     override fun write(ctx: ChannelHandlerContext, msg: Any, promise: ChannelPromise) {
         when (msg) {
+            is DatagramPacketWithContext -> {
+                write(msg, promise, ctx)
+                if (msg.sessionContext.sessionExpirationHint) {
+                    promise.toCompletableFuture().whenComplete { isSuccess, err ->
+                        if (err == null && isSuccess) {
+                            dtlsServer.closeSession(msg.recipient())
+                        }
+                    }
```

Comment:
This could be simplified:

```suggestion
                    promise.toCompletableFuture().thenAccept { dtlsServer.closeSession(msg.recipient()) }
```